Xamarin.Essentials/TextToSpeech/TextToSpeech.mac.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using AppKit;
+
+namespace Xamarin.Essentials
+{
+    public static partial class TextToSpeech
+    {
+        static readonly Lazy<NSSpeechSynthesizer> speechSynthesizer = new Lazy<NSSpeechSynthesizer>(() =>
+            new NSSpeechSynthesizer { Delegate = new SpeechSynthesizerDelegate() });
+
+        internal static Task<IEnumerable<Locale>> PlatformGetLocalesAsync() =>
+            Task.FromResult(NSSpeechSynthesizer.AvailableVoices
+                .Select(v => new Locale(v, null, null, null)));
+
+        internal static async Task PlatformSpeakAsync(string text, SpeechOptions options, CancellationToken cancelToken = default)
+        {
+            var ss = speechSynthesizer.Value;
+            var ssd = (SpeechSynthesizerDelegate)ss.Delegate;
+
+            var tcs = new TaskCompletionSource<bool>();
+            try
+            {
+                // Ensures linker doesn't remove.
+                if (DateTime.UtcNow.Ticks < 0)
+                {
+                    new NSSpeechSynthesizer();
+                    new SpeechSynthesizerDelegate();
```

Comment:
I don't think you need this 2nd one since the earlier
```
new NSSpeechSynthesizer { Delegate = new SpeechSynthesizerDelegate() });
```
should have done the job